core/src/main/java/bisq/core/btc/setup/WalletsSetup.java

```diff
@@ -220,19 +219,12 @@ protected void onSetupCompleted() {
                 if (preferences.getBitcoinNodes() != null && !preferences.getBitcoinNodes().isEmpty())
                     peerGroup.setAddPeersFromAddressMessage(false);
 
-                peerGroup.addConnectedEventListener((peer, peerCount) -> {
-                    // We get called here on our user thread
-                    numPeers.set(peerCount);
-                    connectedPeers.set(peerGroup.getConnectedPeers());
-                });
-                peerGroup.addDisconnectedEventListener((peer, peerCount) -> {
-                    // We get called here on our user thread
-                    numPeers.set(peerCount);
-                    connectedPeers.set(peerGroup.getConnectedPeers());
-                });
-                peerGroup.addBlocksDownloadedEventListener((peer, block, filteredBlock, blocksLeft) -> {
-                    blocksDownloadedFromPeer.set(peer);
-                });
+                UserThread.runPeriodically(() -> {
+                    List<MoneroDaemonPeer> peers = vXmrDaemon.getKnownPeers().stream().filter(MoneroDaemonPeer::isOnline).collect(Collectors.toList());
```

Comment:
Tracked my issue to having zero known peers.

Instead of using `getKnownPeers().get(i).isOnline()`, use `getConnections().get(i).getPeer().isOnline()`. This should reflect the current connections instead of all previously known connections.

Reply:
I changed this & rebased on master. Tests passed for me.
Do you know why getKnownPeers returned something else? are those previously connected peers that were online (?) but not connected to us anymore?

Reply:
`getKnownPeers()` calls daemon rpc `get_peer_list` which indicates when the peers were last seen, not if they're currently online.

`getConnections()` calls daemon rpc `get_connections` which retrieves current incoming and outgoing connections to the node.